### PR TITLE
21 feed page

### PR DESCRIPTION
**Pull request by AlaaTaima**
Relates #21 
Feed-page :
At this page the user can filter his entries depends on their month.
so, it as a history page that the user can find all his entries here.

### CODE REVIEW

File: src/containers/Feed/feeds.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-plusplus */
```
Comment (Fatmasiam) [inline src/containers/Feed/feeds.jsx]:
```suggestion
```

File: src/containers/Feed/data.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    date: 'July',
+  },
+  {
+    grateful: 'family',
+    challenge: 'no time',
+    developing: 'care alot about my social life',
+    time: '07:00',
+    date: 'July',
+  },
+  {
```
Comment (Fatmasiam) [inline src/containers/Feed/data.js]:
```js
{
 greatefull: { 
        title: 'family'
        body: 'example body text for displaying only'
     },
 challenges: { 
        title: 'no time',
        body: 'example body text for displaying only'
     },
   timestamp : 'Tue Nov 05 2019 02:06:03 GMT+0200'
}
```

Comment (Fatmasiam) [reply src/containers/Feed/data.js]: refactor your code according to these data

File: src/containers/Feed/feeds.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                <span>{element.count}</span>
+              </Option>
+            ))}
+          </Select>
+        </div>
+
+        {data.length > 0 ? (
+          data.map(journal => (
+            <JournalCard
+              time={journal.time}
```
Comment (Fatmasiam) [inline src/containers/Feed/feeds.jsx]: this is giving error on console that needs a key

File: src/containers/Feed/feeds.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+import JournalCard from '../../components/JournalCard';
+
+import { months, fakeData } from './data';
+
+import './feeds.css';
+import LogoHeader from '../../components/LogoHeader';
+
+const { Option } = Select;
+class Feed extends Component {
```
Comment (yosefanajjar) [inline src/containers/Feed/feeds.jsx:13]: why are you using a constructor?

File: src/containers/Feed/feeds.jsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    const keys = Object.keys(filteredObject);
+
+    for (let i = 0; i < months.length; i++) {
+      for (let j = 0; j < keys.length; j++) {
+        if (months[i].month === keys[j]) {
+          months[i].count = filteredObject[keys[j]];
+        }
+      }
+    }
+    return this.setState({ monthCount: months });
```
Comment (tomduggan85) [inline src/containers/Feed/feeds.jsx]: @AlaaTaima I think you don't need `return` here.

**Review by fares98 — DISMISSED** [review]

File: src/containers/Feed/feeds.jsx (hunk 5 above)
Comment (AlaaTaima) [reply src/containers/Feed/feeds.jsx]: Yea, you are right , thanks for your comment .

**Review by yosefanajjar — APPROVED** [review]

**Review by fares98 — APPROVED** [review]

File: src/containers/Feed/feeds.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    //
+    const monthArray = fakeData.map(journal =>
+      moment(journal.timestamp).format('MMMM')
+    );
+    const filteredObject = monthArray.reduce((acc, curr) => {
+      if (typeof acc[curr] == 'undefined') {
+        acc[curr] = 1;
+      } else {
+        acc[curr] += 1;
+      }
```
Comment (tomduggan85) [inline src/containers/Feed/feeds.jsx:34]: @AlaaTaima If you want, I think you could shorten this part (lines 30-34) to this:
`acc[curr] = (acc[curr] || 0) + 1`

Comment (tomduggan85) [reply src/containers/Feed/feeds.jsx:34]: .. But it's not super important, just shorter :)

**Comment by tomduggan85** [comment]
@AlaaTaima Overall it looks good to me!

**Review by alaa-yasin — APPROVED** [review]

**Review by Fatmasiam — APPROVED** [review]

**Pull request merged**